akka-docs/src/main/paradox/typed/distributed-pub-sub.md

```diff
@@ -1,91 +1,66 @@
-
-Scala
-:  @@snip [DistributedPubSubExample.scala](/akka-cluster-typed/src/test/scala/docs/akka/cluster/typed/DistributedPubSubExample.scala) { #mediator }
+The topic actor needs to run on each node where subscribers will live or that wants to publish messages to the topic, but
+there is no requirements
  
-Actors register to a topic for Pub-Sub mode, or register to a path for point-to-point mode. 
-
-## Publish
-
-Pub-Sub mode. For the full Akka Classic documentation of this feature see @ref:[Classic Distributed PubSub Publish](../distributed-pub-sub.md#publish).
+The identity of the topic is a tuple of the type of messages that can be published and a string topic name but it is recommended
+to not define multiple topics with different types and the same topic name.
 
-### Subscribers
+Scala
+:  @@snip [PubSubExample.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/pubsub/PubSubExample.scala) { #start-topic }
 
-Subscriber actors can be started on several nodes in the cluster, and all will receive
-messages published to the "content" topic. 
+Java
+:  @@snip [PubSubExample.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/pubsub/PubSubExample.java) { #start-topic }
 
-An actor that subscribes to a topic:
+Local actors can then subscribe to the topic (and unsubscribe from it):
 
 Scala
-:  @@snip [DistributedPubSubExample.scala](/akka-cluster-typed/src/test/scala/docs/akka/cluster/typed/DistributedPubSubExample.scala) { #subscriber }
+:  @@snip [PubSubExample.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/pubsub/PubSubExample.scala) { #subscribe }
 
+Java
+:  @@snip [PubSubExample.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/pubsub/PubSubExample.java) { #subscribe }
 
-Actors may also be subscribed to a named topic with a `group` id. 
-For the full feature description see @ref:[topic groups](../distributed-pub-sub.md#topic-groups).
+And publish messages to the topic:
 
-### Publishers
+Scala
+:  @@snip [PubSubExample.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/pubsub/PubSubExample.scala) { #publish }
 
-Publishers publish messages to the topic from anywhere in the cluster.
-Messages are published by sending `DistributedPubSubMediator.Publish` message to the
-local mediator.
+Java
+:  @@snip [PubSubExample.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/pubsub/PubSubExample.java) { #publish }
 
-An actor that publishes to a topic:
+## Pub Sub Scalability
 
-Scala
-:  @@snip [DistributedPubSubExample.scala](/akka-cluster-typed/src/test/scala/docs/akka/cluster/typed/DistributedPubSubExample.scala) { #publisher }
+Each topic is represented by one @ref[Receptionist](actor-discovery.md) service key meaning that the number of topics 
+will scale to thousands or tens of thousands but for higher numbers of topics will require custom solutions.
```

Comment:
also mention "high turnaround" limitation. we have something about in receptionist docs and it's same kind of limit

Reply:
It actually isn't because the high turnaround would be for subscribers which are local only, the problem would only be with a very high rate of going back and forth between 0 and >0 subscribers which would cause the topic to continuously deregister and register with the receptionist.

Reply:
I mean turnaround of new topic names. I have seen crazy usage of classic pub sub for that. Creating a new topic for each request-response interaction.

Reply:
Ah, good point. I'll clarify